testing/kfctl/endpoint_ready_test.py

```diff
@@ -28,11 +28,12 @@ def test_endpoint_is_ready(record_xml_attribute, project, app_name):
   """
   util.set_pytest_junit(record_xml_attribute, "test_endpoint_is_ready")
 
-  # Owned by project kubeflow-ci-deployment.
-  os.environ["CLIENT_ID"] = "29647740582-7meo6c7a9a76jvg54j0g2lv8lrsb4l8g.apps.googleusercontent.com"
-  if not gcp_util.endpoint_is_ready(
-      "https://{}.endpoints.{}.cloud.goog".format(app_name, project),
-      wait_min=25):
+  if not use_basic_auth:
```

Comment:
Don't we need to provide username and password in the case of basic auth?

Reply:
that depends on whether you'd like to test kflogin with POST.

Reply:
Can we test that if we send a request with basic auth headers containing username/password that we can access the central dashboard? That would probably be my P0.

Testing kflogin would also be good. Is it possible to do both?

I think you will need to modify kfctl_go_build_deploy.py to save the username and password in the test directory (make sure to save it in a location that doesn't get uploaded to GCS for prow).

Reply:
I think you are talking about the same thing, tho.  it's actually sending username/pw to kflogin to get the auth token and then we could access the central dashboard.

Reply:
updated, but you might wanna see if I write to the correct directory - I write to `app_path`

Reply:
Great thanks! Is the test just checking that we get a 200?That's probably good enough for now but is there someway we could verify we actually get the centraldashboard?

Reply:
the content shows up as central dashboard.  but I'm not sure if it's a safe way to find a substr of that as it might be changed.